dm/templates/vpn/vpn.py

```diff
@@ -21,19 +21,34 @@ def generate_config(context):
     properties = context.properties
     project_id = properties.get('project', context.env['project'])
 
-    network = generate_network_url(
+    network = context.properties.get('networkURL', generate_network_uri(
         project_id,
-        context.properties['network']
-    )
+        context.properties.get('networkName','')
+    ))
     target_vpn_gateway = context.env['name'] + '-tvpng'
-    static_ip = context.env['name'] + '-ip'
     esp_rule = context.env['name'] + '-esp-rule'
     udp_500_rule = context.env['name'] + '-udp-500-rule'
     udp_4500_rule = context.env['name'] + '-udp-4500-rule'
     vpn_tunnel = context.env['name'] + '-vpn'
     router_vpn_binding = context.env['name'] + '-router-vpn-binding'
+    resources = []
+    static_ip = ''
+    ip_address = ''
+    if 'ipAddress' in context.properties:
+        ip_address = context.properties['ipAddress']
+    else:
+        static_ip = context.env['name'] + '-ip'
+        resources.append({
+            # The reserved address resource.
+            'name': static_ip,
+            'type': 'gcp-types/compute-v1:addresses',
+            'properties': {
+                'region': context.properties['region']
```

Comment:
Don't we still need a name and project specified?

Reply:
Based on my test, the "addresses" API works with or without "project". I can add the "project" back to keep the old implementation.

Reply:
If you don't specify project, it will use the DM project which isn't always desired.